File: androidApp/src/main/kotlin/com/malaria/android/ui/onboarding/components/WizardStepContainer.kt
package com.malaria.android.ui.onboarding.components

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.Button
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedButton
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp

/**
 * Consistent layout for every onboarding wizard step (spec §10).
 *
 * Mirrors `iosApp/Views/Onboarding/Components/WizardStepContainer.swift`:
 * title + scrollable body + primary button pinned to the bottom, optional
 * secondary action above primary.
 *
 * RTL-readiness (spec §11): uses `Modifier.padding(start = ..., end = ...)`
 * exclusively. Never `left` / `right`.
 */
@Composable
fun WizardStepContainer(
    title: String,
    primaryLabel: String,
    onPrimary: () -> Unit,
    modifier: Modifier = Modifier,
    subtitle: String? = null,
    stepIndicator: String? = null,
    primaryEnabled: Boolean = true,
    secondaryLabel: String? = null,
    onSecondary: (() -> Unit)? = null,
    body: @Composable () -> Unit,
) {
    Surface(modifier = modifier.fillMaxSize()) {
        Column(modifier = Modifier.fillMaxSize()) {
            if (stepIndicator != null) {
                Text(
                    text = stepIndicator,
                    style = MaterialTheme.typography.labelSmall,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                    modifier = Modifier.padding(
                        top = 24.dp,
                        start = 24.dp,
                        end = 24.dp,
                    ),
                )
            }
            Text(
                text = title,
                style = MaterialTheme.typography.headlineMedium,
                modifier = Modifier.padding(
                    top = if (stepIndicator == null) 24.dp else 8.dp,
                    start = 24.dp,
                    end = 24.dp,
                ),
            )
            if (subtitle != null) {
                Text(
                    text = subtitle,
                    style = MaterialTheme.typography.bodyMedium,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                    modifier = Modifier.padding(
                        top = 8.dp,
                        start = 24.dp,
                        end = 24.dp,
                    ),
                )
            }

            Box(
                modifier = Modifier
                    .weight(1f)
                    .fillMaxWidth(),
            ) {
                Column(
                    modifier = Modifier
                        .verticalScroll(rememberScrollState())
                        .padding(
                            top = 24.dp,
                            start = 24.dp,
                            end = 24.dp,
                            bottom = 16.dp,
                        ),
                    verticalArrangement = Arrangement.spacedBy(16.dp),
                ) {
                    body()
                }
            }

            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(
                        start = 24.dp,
                        end = 24.dp,
                        top = 8.dp,
                        bottom = 32.dp,
                    ),
                verticalArrangement = Arrangement.spacedBy(8.dp),
            ) {
                if (secondaryLabel != null && onSecondary != null) {
                    OutlinedButton(
                        onClick = onSecondary,
                        modifier = Modifier.fillMaxWidth(),
                        contentPadding = PaddingValues(vertical = 12.dp),
                    ) {
                        Text(secondaryLabel)
                    }
                }
                Button(
                    onClick = onPrimary,
                    enabled = primaryEnabled,
                    modifier = Modifier.fillMaxWidth(),
                    contentPadding = PaddingValues(vertical = 12.dp),
                ) {
                    Text(primaryLabel)
                }
            }
        }
    }
}

File: iosApp/Views/Onboarding/Components/WizardStepContainer.swift
import SwiftUI

/// Consistent layout for every onboarding wizard step (spec §10).
///
/// All steps share the same skeleton: title + scrollable body + primary
/// action button pinned to the bottom, with an optional secondary action
/// rendered above the primary one.
///
/// RTL-readiness (spec §11): uses `.leading` / `.trailing` padding and
/// alignment exclusively. Never `.left` / `.right`.
struct WizardStepContainer<Body: View>: View {

    let title: String
    let subtitle: String?
    let primaryAction: PrimaryAction
    let secondaryAction: SecondaryAction?
    let stepIndicator: String?
    @ViewBuilder var content: () -> Body

    init(
        title: String,
        subtitle: String? = nil,
        stepIndicator: String? = nil,
        primary: PrimaryAction,
        secondary: SecondaryAction? = nil,
        @ViewBuilder content: @escaping () -> Body
    ) {
        self.title = title
        self.subtitle = subtitle
        self.stepIndicator = stepIndicator
        self.primaryAction = primary
        self.secondaryAction = secondary
        self.content = content
    }

    var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            if let stepIndicator {
                Text(stepIndicator)
                    .font(.caption)
                    .foregroundStyle(.secondary)
                    .padding(.top, 24)
                    .padding(.leading, 24)
                    .padding(.trailing, 24)
            }
            Text(title)
                .font(.largeTitle.weight(.semibold))
                .padding(.top, stepIndicator == nil ? 24 : 8)
                .padding(.leading, 24)
                .padding(.trailing, 24)
            if let subtitle {
                Text(subtitle)
                    .font(.body)
                    .foregroundStyle(.secondary)
                    .padding(.top, 8)
                    .padding(.leading, 24)
                    .padding(.trailing, 24)
            }

            ScrollView {
                VStack(alignment: .leading, spacing: 16) {
                    content()
                }
                .padding(.top, 24)
                .padding(.leading, 24)
                .padding(.trailing, 24)
                .padding(.bottom, 16)
                .frame(maxWidth: .infinity, alignment: .leading)
            }

            VStack(spacing: 8) {
                if let secondaryAction {
                    Button(action: secondaryAction.handler) {
                        Text(secondaryAction.title)
                            .frame(maxWidth: .infinity)
                    }
                    .buttonStyle(.bordered)
                    .controlSize(.large)
                }
                Button(action: primaryAction.handler) {
                    Text(primaryAction.title)
                        .frame(maxWidth: .infinity)
                }
                .buttonStyle(.borderedProminent)
                .controlSize(.large)
                .disabled(!primaryAction.isEnabled)
            }
            .padding(.leading, 24)
            .padding(.trailing, 24)
            .padding(.bottom, 32)
            .padding(.top, 8)
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .leading)
    }
}

extension WizardStepContainer {
    struct PrimaryAction {
        let title: String
        let isEnabled: Bool
        let handler: () -> Void

        init(title: String, isEnabled: Bool = true, handler: @escaping () -> Void) {
            self.title = title
            self.isEnabled = isEnabled
            self.handler = handler
        }
    }

    struct SecondaryAction {
        let title: String
        let handler: () -> Void
    }
}
